resources/linux/debian/postinst.template

```diff
@@ -35,19 +35,42 @@ if [ "@@NAME@@" != "code-oss" ]; then
 	eval $(apt-config shell APT_TRUSTED_PARTS Dir::Etc::trustedparts/d)
 	CODE_TRUSTED_PART=${APT_TRUSTED_PARTS}microsoft.gpg
 
+	RET=true
+	if [ -e '/usr/share/debconf/confmodule' ]; then
+		. /usr/share/debconf/confmodule
+		db_get @@NAME@@/add-microsoft-repo || true
+	fi
+
 	# Install repository source list
 	WRITE_SOURCE=0
-	if [ ! -f $CODE_SOURCE_PART ] && [ ! -f /etc/rpi-issue ]; then
-		# Write source list if it does not exist and we're not running on Raspberry Pi OS
-		WRITE_SOURCE=1
-	elif grep -Eq "http:\/\/packages\.microsoft\.com\/repos\/vscode" $CODE_SOURCE_PART; then
+	if [ "$RET" false ]; then
+		# The user does not want to add the microsoft repository
+		WRITE_SOURCE=0
+	elif grep -q "http://packages.microsoft.com/repos/vscode" $CODE_SOURCE_PART; then
 		# Migrate from old repository
 		WRITE_SOURCE=1
+	elif apt-cache policy | grep -q "https://packages.microsoft.com/"; then
+		# Skip following checks if the repo is already known to apt
+		WRITE_SOURCE=0
+	elif [ ! -f $CODE_SOURCE_PART ] && [ ! -f /etc/rpi-issue ]; then
+		# Write source list if it does not exist and we're not running on Raspberry Pi OS
+		WRITE_SOURCE=1
 	elif grep -q "# disabled on upgrade to" $CODE_SOURCE_PART; then
 		# Write source list if it was disabled by OS upgrade
 		WRITE_SOURCE=1
 	fi
 
+	if [ "$WRITE_SOURCE" -eq "1" ] && [ -e '/usr/share/debconf/confmodule' ]; then
```

Comment:
Could you add another variable to check whether we're migrating from http://packages.microsoft.com/repos/vscode or http://packages.microsoft.com/repos/code to https://packages.microsoft.com/repos/code? I believe that migrations don't have to go through this flow.

Reply:
Fair point, I used value `2` for migrations. That way, is is possible to ignore `2` for debconf and to then check `-ne 0` for the actual write. Anyone not wanting to have the source repo added to their system will most likely not have that file around at all.